pkg/cloud/cloud.go

```diff
@@ -227,15 +228,21 @@ var _ Cloud = &cloud{}
 
 // NewCloud returns a new instance of AWS cloud
 // It panics if session is invalid
-func NewCloud(region string) (Cloud, error) {
-	return newEC2Cloud(region)
+func NewCloud(region string, awsAssumeRole string) (Cloud, error) {
+	return newEC2Cloud(region, awsAssumeRole)
 }
 
-func newEC2Cloud(region string) (Cloud, error) {
+func newEC2Cloud(region string, awsAssumeRole string) (Cloud, error) {
 	awsConfig := &aws.Config{
 		Region:                        aws.String(region),
 		CredentialsChainVerboseErrors: aws.Bool(true),
 	}
+	sess := session.Must(session.NewSession(awsConfig))
+
+	if awsAssumeRole != "" {
```

Comment:
What's the behavior if this string is not a valid ARN? I'd say we should verify and if it's not, we should return an error and fail the startup. 

Can we add a test case for that?

Reply:
Well... it seems to do nothing until it actually tries to exercise it? (I'm surprised)

    cloud.go:243] Assuming role into ARN foo

Reply:
We should definitely verify it then and fail early. Can we add a check here? 

Ref: https://docs.aws.amazon.com/sdk-for-go/api/aws/arn/#IsARN

Reply:
Added